lazy_theta_star_planner/src/lazy_theta_star_planner.cpp

```diff
@@ -0,0 +1,153 @@
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmapRos)
+{
+  planner_ = std::make_unique<lazyThetaStar::LazyThetaStar>();
+  node_ = parent;
+  name_ = name;
+  planner_->costmap_ = costmapRos->getCostmap();
+  globalFrame_ = costmapRos->getGlobalFrameID();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".how_many_corners", rclcpp::ParameterValue(
+      8));
+
+  node_->get_parameter(name_ + ".how_many_corners", how_many_corners_);
+
+  if (how_many_corners_ != 8) {
+    how_many_corners_ = 4;
+  }
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".interpolation_dist", rclcpp::ParameterValue(
+      0.1));
+
+  node_->get_parameter(name_ + ".interpolation_dist", interpolation_dist_);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".lethal_cost", rclcpp::ParameterValue(
+      127));
+
+  node_->get_parameter(name_ + ".lethal_cost", lethal_cost_);
+}
+
+void LazyThetaStarPlanner::cleanup()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "CleaningUp plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+void LazyThetaStarPlanner::activate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Activating plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+void LazyThetaStarPlanner::deactivate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Deactivating plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+nav_msgs::msg::Path LazyThetaStarPlanner::createPlan(
+  const geometry_msgs::msg::PoseStamped & start,
+  const geometry_msgs::msg::PoseStamped & goal)
+{
+  nav_msgs::msg::Path global_path;
+  std::vector<coords<world_pts>> path;
+
+  planner_->costmap_->worldToMap(start.pose.position.x, start.pose.position.y, src_[0], src_[1]);
+  planner_->src = {static_cast<int>(src_[0]), static_cast<int>(src_[1])};
+
+  planner_->costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, dst_[0], dst_[1]);
+  planner_->dst = {static_cast<int>(dst_[0]), static_cast<int>(dst_[1])};
+
+  RCLCPP_INFO(
+    node_->get_logger(), "Got the src and dst... (%i, %i) && (%i, %i)",
+    planner_->src.x, planner_->src.y, planner_->dst.x, planner_->dst.y);
+
+  planner_->node_ = node_;
+  planner_->how_many_corners = how_many_corners_;
+
+  planner_->initializeStuff();
+  planner_->getPath(path);
+  planner_->clearStuff();
+
+  RCLCPP_INFO(node_->get_logger(), "RECEIVED THE PATH");
+
+  if (path.size() > 1) {
+    for (int i = 0; i < path.size(); i++) {
+      std::cout << path[i].x << '\t' << path[i].y << '\n';
+    }
+
+    global_path = linearInterpolation(path, 0.1);
+  } else {
+    global_path.poses.clear();
+  }
+  path.clear();
+  std::cout << global_path.poses.size() << '\n';
+  global_path.header.stamp = node_->now();
+  global_path.header.frame_id = globalFrame_;
+  RCLCPP_INFO(node_->get_logger(), "SENT OUT THE PATH");
+  return global_path;
+}
+
+nav_msgs::msg::Path LazyThetaStarPlanner::linearInterpolation(
```

Comment:
Can you explain why you added this? Are the points that sparse? Maybe its worth having it be uniformly sampled rather than linearly sampled if the distance between points can vary

Reply:
The raw output of the algorithm are indeed sparse points which are in line of sight of each other, so I have to interpolate, to generate points in between them.

Reply:
Are they uniformly spaced (I assume not)? If so, maybe we should literally interpolate by costmap cell size rather than just splitting the difference so they are uniformly spaced.

Reply:
Alright

Reply:
It is uniformly spaced, with distance between each point being equal to the “.interpolation_dist” provided by the user

Reply:
OK